rpmlb/builder/koji.py

```diff
@@ -0,0 +1,173 @@
+"""Builder interface for the Koji build service."""
+
+import re
+from pathlib import Path
+from typing import Iterator, Optional, Sequence
+
+from ..utils import run_cmd, run_cmd_with_capture
+from ..work import Work
+from .base import BaseBuilder
+
+
+class KojiBuilder(BaseBuilder):
+    """Builder interface for the Koji build service."""
+
+    DEFAULT_TARGET_FORMAT = 'sclo{epel}-{collection}-rh-el{epel}'
+
+    def __init__(self):
+        """Initialize the slots"""
+
+        self.collection = None  #: The collection to build
+        self.epel = None  #: Version of EPEL to build for
+        self.profile = None  #: The koji configuration profile to use
+        #: The format string for construction of the actual template name
+        self.target_format = type(self).DEFAULT_TARGET_FORMAT
+
+    @property
+    def base_command(self) -> Sequence[str]:
```

Comment:
Your question is why is following a typical design pattern such as template pattern beneficial?  
Hmm, the reason is same with why the template pattern exists.

It may be a little impolite for you.
Have you developed big application or service with a object oriented design?